tests/algorithms/test_algorithm_resumption.py

```diff
@@ -0,0 +1,236 @@
+        return
+
+    if model_name is None and algo_name in simple_net_skiplist:
+        pytest.skip(simple_net_skiplist[algo_name])
+
+    if model_name == 'resnet50_synthetic' and algo_name in resnet_skiplist:
+        pytest.skip(resnet_skiplist[algo_name])
+    
+    if model_name == 'gpt2_52m' and algo_name in lm_skiplist:
+        pytest.skip(lm_skiplist[algo_name])
+
+    if algo_name in algo_hparams_overrides:
+        algo_object = algo_hparams(**algo_hparams_overrides[algo_name])
+    else:
+        algo_object = algo_hparams()
+
+    composer_trainer_hparams.algorithms = [algo_object]
+    composer_trainer_hparams.save_name_format = save_name_format
+    composer_trainer_hparams.train_dataset.use_synthetic = True
+    composer_trainer_hparams.train_dataset.synthetic_num_unique_samples = 500
+    composer_trainer_hparams.train_dataset.shuffle = False
+    composer_trainer_hparams.val_dataset.use_synthetic = True
+    composer_trainer_hparams.val_dataset.shuffle = False
+    composer_trainer_hparams.grad_accum = 2
+    composer_trainer_hparams.loggers = []
+    composer_trainer_hparams.train_batch_size = 10
+    composer_trainer_hparams.eval_batch_size = 10
+    num_epochs = 5
+    composer_trainer_hparams.max_duration = f"{num_epochs}ep"
+    composer_trainer_hparams.precision = Precision.FP32
+    composer_trainer_hparams.train_subset_num_batches = 5
+    composer_trainer_hparams.eval_subset_num_batches = 5
+    composer_trainer_hparams.device = device_hparams
+    if deepspeed_enabled:
+        assert zero_stage is not None
+        if zero_stage > 0:
+            composer_trainer_hparams.deterministic_mode = False
+            if model_name is not None:
+                pytest.skip(
+                    textwrap.dedent(f"""\
+                        Skipping test since deterministic mode is required for
+                        non-trivial models, but deterministic mode isn't compatible with deepspeed
+                        zero stage {zero_stage}"""))
+        composer_trainer_hparams.deepspeed = {"zero_optimization": {"stage": zero_stage}}
+
+    checkpoint_a_folder = "first"
+    composer_trainer_hparams.save_folder = checkpoint_a_folder
+    composer_trainer_hparams.save_interval = save_interval
+    composer_trainer_hparams.seed = seed
+
+    composer_trainer_hparams.validate_every_n_batches = 1 if resume_file.startswith("ba") else 0
+    composer_trainer_hparams.validate_every_n_epochs = 1 if resume_file.startswith("ep") else 0
+    first_trainer = composer_trainer_hparams.initialize_object()
+    first_trainer.fit()
+    save_interval_time = Time.from_timestring(save_interval)
+    if save_interval_time.unit == TimeUnit.EPOCH:
+        expected_num_checkpoints = ((num_epochs - 1) // save_interval_time.value) + 1
+    else:
+        expected_num_checkpoints = (
+            (composer_trainer_hparams.train_subset_num_batches * num_epochs - 1) // save_interval_time.value) + 1
```

Comment:
Since this path is tested elsewhere, let's remove this code and the asserts below. The unit tests should just test one thing -- algorithm resumption.